modules/dreambooth/dreambooth.py

```diff
@@ -0,0 +1,789 @@
+              "seed": seed, "grad_acc_steps": grad_acc_steps, "warmup_steps": warmup_steps}
+    print("Loading model?")
+    data = TrainConfig().from_file(model_name)
+    values = []
+    if data is not None:
+        if not "db_initialization_text" in data:
+            shared.state.textinfo = "Please train your model before loading config."
+        else:
+            shared.state.textinfo = "Loaded config data."
+        for key in data:
+            value = data[key]
+            print(f"Loading from existing {key}:{value}")
+            output[key] = value
+
+        for key in output:
+            value = output[key]
+            if key != "model_name":
+                print(f"Appending {key}: {value}")
+                values.append(value)
+
+    return values
+
+
+class DreamBooth:
+    # TODO: Clean up notes below and make them a proper docstring
+    def __init__(self,
+                 config: TrainConfig
+                 ):
+        self.model_name = config["model_name"]
+        self.model_dir = config["model_dir"]
+        self.src = config["src"]
+        self.total_steps = config["total_steps"]
+        self.instance_data_dir = config["instance_data_dir"]
+        self.instance_prompt = config["instance_prompt"]
+        self.class_prompt = config["class_prompt"]
+        seed = config["seed"]
+        if seed == -1:
+            self.seed = None
+        else:
+            self.seed = seed
+        self.train_batch_size = config["batch_size"]
+        self.sample_batch_size = config["class_batch_size"]
+        self.max_train_steps = config["steps"]
+        self.num_class_images = config["num_class_images"]
+        self.gradient_accumulation_steps = config["grad_acc_steps"]
+        self.learning_rate = config["learn_rate"]
+        self.lr_scheduler = config["scheduler"]
+        self.lr_warmup_steps = config["warmup_steps"]
+        self.use_8bit_adam = config["use_adam"]
+        self.log_interval = 10
+        self.save_preview_every = config["save_preview_every"]
+        self.save_data_every = config["save_embedding_every"]
+        self.mixed_precision = config["mixed_precision"]
+
+        name = "".join(x for x in config["model_name"] if x.isalnum())
+        model_path = paths.models_path
+        model_dir = os.path.join(model_path, "dreambooth", name)
+        self.output_dir = os.path.join(model_dir, "working")
+        # A folder containing the training data of instance images.
+        if config["class_data_dir"] is not None and config["class_data_dir"] != "":
+            self.class_data_dir = config["class_data_dir"]
+        else:
+            self.class_data_dir = os.path.join(model_dir, "classifiers")
+            if not os.path.exists(self.class_data_dir):
+                os.makedirs(self.class_data_dir)
+
+        self.logging_dir = os.path.join(self.output_dir, "logging")
+        self.pretrained_model_path = os.path.join(model_dir, "stable-diffusion-v1-5")
```

Comment:
Does every model require its own copy of `stable-diffusion-v1-5`? Can it be downloaded just once to `models/dreambooth` instead?

Reply:
Technically, *no* model requires a copy. The files are extracted from the target checkpoint when you create a new dreambooth model. They will reside on disk until training is completed and the datafolder for that model is deleted manually.

The only file that actually gets downloaded is the config file needed to load the model.